File: CPE/Exam/220322/pC_UVA-13171.py
""" UVA-13171: Pixel Art
    模擬(Simulation)
    已於 UVA, CPE, ZeroJudge 測試通過
"""
t = int(input())

for _ in range(t):
    m, y, c, s = input().split()
    m, y, c = map(int, [m, y, c]) # magenta, yellow, cyan
    flag = True
    for ch in s:
        if ch == 'M':
            m -= 1
        elif ch == 'Y':
            y -= 1
        elif ch == 'C':
            c -= 1
        elif ch == 'R':
            m -= 1
            y -= 1
        elif ch == 'V':
            m -= 1
            c -= 1
        elif ch == 'G':
            y -= 1
            c -= 1
        elif ch == 'B':
            m -= 1
            y -= 1
            c -= 1
        if m < 0 or y < 0 or c < 0:
            flag = False
            break
    print(f"YES {m} {y} {c}" if flag else 'NO')

File: CPE/Exam/220322/pC_UVA-13171.cpp
/* UVA-13171: Pixel Art
    模擬(Simulation)
    已於 UVA, CPE, ZeroJudge 測試通過
*/
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int t;
    cin >> t;
    while (t--) {
        int m, y, c;
        string s;
        cin >> m >> y >> c >> s;
        bool flag = true;
        for (char ch : s) {
            if (ch == 'M') {
                m -= 1;
            } else if (ch == 'Y') {
                y -= 1;
            } else if (ch == 'C') {
                c -= 1;
            } else if (ch == 'R') {
                m -= 1;
                y -= 1;
            } else if (ch == 'V') {
                m -= 1;
                c -= 1;
            } else if (ch == 'G') {
                y -= 1;
                c -= 1;
            } else if (ch == 'B') {
                m -= 1;
                y -= 1;
                c -= 1;
            }
            if (m < 0 || y < 0 || c < 0) {
                flag = false;
                break;
            }
        }
        if (flag) {
            cout << "YES " << m << " " << y << " " << c << endl;
        } else {
            cout << "NO" << endl;
        }   
    }
    return 0;
}